### PR TITLE
[TIMOB-23305] Ti.UI.Label width is not updated when setting new text

### PR DESCRIPTION
**Pull request by infosia**
[TIMOB-23305](https://jira.appcelerator.org/browse/TIMOB-23305)

Attempt to update width when setting new text. Currently it's blocked because Xaml `TextBlock` does not shrink its size with shorter text.

* `TextBlock` does not change its size when `TextWrapping` is `Wrap`.
* `TextBlock` expands its size when `TextWrapping` is `NoWrap`.
* `TextBlock` does not shrink its size in every cases of text wrapping.
* ~~We might need to find a way to calculate its size for the given text and font manually.~~ <- `Measure()` works good for it

```javascript
var win = Ti.UI.createWindow({
    backgroundColor: 'green',
    layout: 'vertical'
});

var update = true;

var topView = Ti.UI.createView({ backgroundColor: 'blue', width: Ti.UI.FILL, height: '10%' });
var button = Ti.UI.createButton({ title: 'Change Text' });
topView.add(button);

var label = Ti.UI.createLabel({
    text: update ? 'Lorem ipsum' : 'Lorem ipsum dolor sit amet.',
    backgroundColor: 'orange'
});

button.addEventListener('click', function () {
    label.text = update ? 'Lorem ipsum dolor sit amet.' : 'Lorem ipsum';
    update = !update;
});

var bottomView = Ti.UI.createView({backgroundColor:'blue', width:Ti.UI.FILL, height: '10%'});

win.add(topView);
win.add(label);
win.add(bottomView);

win.open();
```

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
It appears that the general solution people are proposing is to create a TextBlock, set the text/font, then call Measure and Arrange on it and grab the ActualWidth/ActualHeigth properties afterwards.

See:
- http://stackoverflow.com/questions/9126076/how-can-i-calculate-the-width-of-a-string-in-metro-without-displaying-it
- https://invokeit.wordpress.com/2012/09/19/how-to-measure-rendered-string-dimensions-in-win8dev/

It may be that we just need to call something like UIElement.UpdateLayout for the parent or something?

**Comment by infosia** [comment]
`Measure()` is working good for TextBlock 👍 Now I'm trying to ensure it doesn't break anything...

**Comment by infosia** [comment]
build win 10

**Comment by infosia** [comment]
build win 10

**Comment by garymathews** [comment]
build win 10

**Comment by garymathews** [comment]
build win 10

File: Source/UI/src/Label.cpp [hunk 1]
```diff
@@ -64,6 +64,18 @@ namespace TitaniumWindows
 
 			const auto layout = getViewLayoutDelegate<WindowsViewLayoutDelegate>();
 
+			layout->postRequestLayoutHook__ = [this, layout]() {
```
Comment (garymathews) [inline Source/UI/src/Label.cpp]: Is this needed, the label seem to scale correctly first time without this?

Comment (infosia) [inline Source/UI/src/Label.cpp]: Oops right it does work without this...I think I was confused. I saw `rect` property is not update so I was to make sure LayoutEngine calculate it again. I'll update to better one.

**Comment by garymathews** [comment]
CR: **PASS**
FR: **PASS**

:+1:

**Comment by infosia** [comment]
build win 10

**Pull request merged**